plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/exec/DBCStatistics.java

```diff
@@ -151,25 +151,27 @@ public boolean isEmpty() {
     }
 
     public void accumulate(DBCStatistics stat) {
-        if (stat.rowsUpdated >= 0) {
-            if (rowsUpdated < 0) rowsUpdated = 0;
-            rowsUpdated += stat.rowsUpdated;
-        }
-        if (stat.rowsFetched > 0) {
-            if (rowsFetched < 0) rowsFetched = 0;
-            rowsFetched += stat.rowsFetched;
-        }
-        executeTime += stat.executeTime;
-        fetchTime += stat.fetchTime;
-        statementsCount += stat.statementsCount;
-        if (!CommonUtils.isEmpty(stat.messages)) {
-            for (String message : stat.messages) {
-                addMessage(message);
+        if (stat != null) {
+            if (stat.rowsUpdated >= 0) {
+                if (rowsUpdated < 0) rowsUpdated = 0;
+                rowsUpdated += stat.rowsUpdated;
             }
-        }
-        if (!CommonUtils.isEmpty(stat.infoMap)) {
-            for (Map.Entry<String, Object> info : stat.infoMap.entrySet()) {
-                addInfo(info.getKey(), info.getValue());
+            if (stat.rowsFetched > 0) {
+                if (rowsFetched < 0) rowsFetched = 0;
+                rowsFetched += stat.rowsFetched;
+            }
+            executeTime += stat.executeTime;
+            fetchTime += stat.fetchTime;
+            statementsCount += stat.statementsCount;
+            if (!CommonUtils.isEmpty(stat.messages)) {
+                for (String message : stat.messages) {
+                    addMessage(message);
+                }
+            }
+            if (!CommonUtils.isEmpty(stat.infoMap)) {
+                for (Map.Entry<String, Object> info : stat.infoMap.entrySet()) {
+                    addInfo(info.getKey(), info.getValue());
+                }

```

Comment:
Let's use an early exit instead of wrapping the entire method with `if` statement.